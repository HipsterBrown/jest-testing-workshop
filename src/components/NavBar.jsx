import React from 'react';
import {Link} from 'redux-little-router';

const NavBar = () => (
  <nav className="bg-near-black ph5 pv3 ttu f5 fw6">
    <Link href="/" className="link underline-hover near-white">
      Home
    </Link>
  </nav>
);

export default NavBar;
