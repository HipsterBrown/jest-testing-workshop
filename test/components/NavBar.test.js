import React from 'react';
import renderer from 'react-test-renderer';
import MockRouterProvider from 'helpers/MockRouterProvider';
import NavBar from 'components/NavBar';

describe('<NavBar />', () => {
  test('displays all links', () => {
    expect(
      renderer
        .create(
          <MockRouterProvider>
            <NavBar />
          </MockRouterProvider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
});
