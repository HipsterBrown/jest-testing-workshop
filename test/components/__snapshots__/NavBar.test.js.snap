// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> displays all links 1`] = `
<nav
  className="bg-near-black ph5 pv3 ttu f5 fw6"
>
  <a
    className="link underline-hover near-white"
    href="/"
    onClick={[Function]}
    target={undefined}
  >
    Home
  </a>
</nav>
`;
